invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/administration/records/CompareRevisions.js

```diff
@@ -0,0 +1,185 @@
+    };
+  }
+
+  componentWillUnmount() {
+    this.cancellableAction && this.cancellableAction.cancel();
+  }
+
+  async componentDidMount() {
+    const { resource } = this.props;
+    this.setState({ loading: true });
+    try {
+      this.cancellableAction = withCancel(fetch(`${resource.links.self}/revisions`));
+      const response = await this.cancellableAction.promise;
+      const revisions = await response.json();
+      this.setState(
+        {
+          allRevisions: revisions,
+          targetRevision: revisions[0],
+          srcRevision: revisions.length > 1 ? revisions[1] : revisions[0],
+        },
+        () => {
+          this.setState({ loading: false });
+        }
+      );
+    } catch (error) {
+      if (error === "UNMOUNTED") return;
+      this.setState({ error: error, loading: false });
+      console.error(error);
+    }
+  }
+
+  static contextType = NotificationContext;
+
+  computeDiff = (before, after) => {
+    const diff = this.differ.diff(before, after);
+    this.setState({ currentDiff: diff });
+  };
+
+  handleModalClose = () => {
+    const { actionCancelCallback } = this.props;
+    actionCancelCallback();
+  };
+
+  render() {
+    const { error, loading, currentDiff, allRevisions, srcRevision, targetRevision } =
+      this.state;
+
+    const options = Object.values(allRevisions).map((rev) => ({
+      key: rev.updated,
+      text: `${rev.updated} (${rev.revision_id})`,
+      value: rev,
+    }));
+
+    return (
+      <>
+        <Grid>
+          <Grid.Column width={6}>
+            <label htmlFor="source-revision">From</label>
+            <Dropdown
+              id="source-revision"
+              loading={loading}
+              placeholder="Source revision..."
+              fluid
+              selection
+              value={srcRevision}
+              onChange={(e, { value }) => {
+                this.setState({ srcRevision: value });
+              }}
+              options={options}
+            />
+          </Grid.Column>
+          <Grid.Column width={1} verticalAlign="middle">
+            <Icon name="exchange"></Icon>
+          </Grid.Column>
+          <Grid.Column width={6}>
+            <label htmlFor="target-revision">To</label>
+            <Dropdown
+              id="target-revision"
+              loading={loading}
+              placeholder="Target revision..."
+              fluid
+              selection
+              value={this.state.targetRevision}
+              onChange={(e, { value }) => {
+                this.setState({ targetRevision: value });
+              }}
+              options={options}
+            />
+          </Grid.Column>
+          <Grid.Column verticalAlign="bottom">
+            <Button
+              onClick={(e, { value }) => {
+                this.computeDiff(srcRevision, targetRevision);
+              }}
+            >
+              Compare
+            </Button>
+          </Grid.Column>
+        </Grid>
+        {error && (
```

Comment:
you should wrap the error in Modal.content too, otherwise it might appear broken